### PR TITLE
HwcLayer: Apply rotation transformation to damage.

### PR DESCRIPTION
**Pull request by harishkrupo**
This enables us to remove the workaround in Overlaylayer.

Jira: None
Test: No regressions seens on Android and Linux.

Signed-off-by: Harish Krupo <harish.krupo.kps@intel.com>

### CODE REVIEW

File: common/core/hwclayer.cpp [hunk 1]
```diff
@@ -325,25 +325,69 @@ bool HwcLayer::IsCursorLayer() const {
   return is_cursor_layer_;
 }
 
+HwcRect<int> HwcLayer::TransformDamage() {
+  HwcRect<int> ret;
+  int ox = 0, oy = 0;
+  HwcRect<int> tr_surface_damage =
```
Comment (kalyankondapally) [inline common/core/hwclayer.cpp]: instead of allocating temp rects, u can use current_Rendering_Damage_ ?

Comment (harishkrupo) [reply common/core/hwclayer.cpp]: I think that would impact the readability of the code? 
I was thinking of removing return value and just assign the ret rect to current_rendering_damage.

Comment (kalyankondapally) [reply common/core/hwclayer.cpp]: hmm no. You can move this chunk of code to the calling function itself I guess to keep it all in one place.

Comment (harishkrupo) [reply common/core/hwclayer.cpp]: okay.

**Comment by kalyankondapally** [comment]
merged

**Pull request closed without merging**